src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -166,4 +185,62 @@ private JsonArray listTemplates(String pageToken, JsonArray jsonElements) throws
     }
     return jsonElements;
   }
+
+  public List<StructuredRecord> listTemplateData(String templateName) throws IOException, AribaException,
+    InterruptedException {
+    AribaServices aribaServices = new AribaServices(config);
+    AribaStructuredTransformer transformer = new AribaStructuredTransformer();
+    try {
+      accessToken = aribaServices.getAccessToken();
+    } catch (AribaException e) {
+      throw new IOException("unable to generate access token", e);
+    }
+    URL dataURL = HttpUrl.parse(config.getBaseURL()).
+      newBuilder()
+      .addPathSegments(METADATA_PATH)
+      .addPathSegments(String.format(DATA_PATH_SEGMENT, config.getSystemType()))
+      .addPathSegments(templateName)
+      .addQueryParameter(ResourceConstants.PRODUCT, ResourceConstants.ANALYTICS)
+      .addQueryParameter(ResourceConstants.REALM, config.getRealm()).build().url();
+    Request request = new Request.Builder().get().url(dataURL).addHeader(API_KEY, config.getApiKey())
+      .addHeader(AUTHORIZATION, aribaServices.getAuthenticationKey(accessToken)).build();
+    OkHttpClient enhancedOkHttpClient = new OkHttpClient();
+    Response response = enhancedOkHttpClient.newCall(request).execute();
+    List<JsonNode> nodeData = new ArrayList<>();
+    List<StructuredRecord> recordList = new ArrayList<>();
```

Comment:
rearrange the lines. Move the variable declaration to the top.
Fetch the schema in starting itself.

Reply:
done